src/app/(app)/leaves/actions/has-leave-request-permission.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  LeaveRequestStatus,
+  MemberRole,
+} from "@/constant/constant";
+import { getLoggedInUser } from "@/utils/getLoggedInUser";
+import { getLeaveRequestById } from "./get-leave-request-by-id";
+
+export const hasLeaveRequestPermission = async (id: number): Promise<boolean> => {
+    const loginUser = await getLoggedInUser();
+    const leaveRequest = await getLeaveRequestById(id);
+  
+    if (!loginUser || !leaveRequest) {
+      return false;
+    }
+  
+    const { role, id: userId } = loginUser;
+    const { status, memberId } = leaveRequest;
+  
+    if (role === MemberRole.Member && status !== LeaveRequestStatus.Pending) {
+      return false;
+    }
+  
+    if (role === MemberRole.ShiftIncharge && memberId === userId && status !== LeaveRequestStatus.Pending) {
+      return false;
+    }
+  
+    if (role === MemberRole.Incharge) {
```

Comment:
@muhammad-bilal-ali-khokhar  We can't use a switch statement because it only handles a single condition type. In the case of the `shift in-charge,` we need to check multiple conditions.

Reply:
In multiple condition's case, we can assign condition to a variable , and that variable can be used in switch ,